lib/Dialect/Pipeline/PipelineOps.cpp

```diff
@@ -246,16 +263,17 @@ static void printPipelineOp(OpAsmPrinter &p, TPipelineOp op) {
 
   // Print the optional stall.
   if (op.hasStall()) {
-    p << "stall(";
-    p.printOperand(op.getStall());
-    p << ") ";
+    printKeywordOperand(p, "stall", op.getStall());
+    p << " ";
   }
 
   // Print the clock, reset, and go.
   printKeywordOperand(p, "clock", op.getClock());
   p << " ";
-  printKeywordOperand(p, "reset", op.getReset());
-  p << " ";
+  if (op.hasReset()) {
+    printKeywordOperand(p, "reset", op.getReset());
```

Comment:
Have you tried round-tripping the IR? I suspect there is an asymmetry in printing vs parsing. I think there's a new flag in circt-opt to automatically round-trip the IR.

Reply:
Round tripping seems to work fine; the existing round-trip tests pass successfully, and they've been extended with the reset signal removal.